File: src/2025-02-23-deepclaude-try/javascript/exercises/practice/house/house.js
//
// This is only a SKELETON file for the 'House' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class House {
  static parts = [
    { subject: 'the house that Jack built.', action: '' },
    { subject: 'the malt', action: 'lay in ' },
    { subject: 'the rat', action: 'ate ' },
    { subject: 'the cat', action: 'killed ' },
    { subject: 'the dog', action: 'worried ' },
    { subject: 'the cow with the crumpled horn', action: 'tossed ' },
    { subject: 'the maiden all forlorn', action: 'milked ' },
    { subject: 'the man all tattered and torn', action: 'kissed ' },
    { subject: 'the priest all shaven and shorn', action: 'married ' },
    { subject: 'the rooster that crowed in the morn', action: 'woke ' },
    { subject: 'the farmer sowing his corn', action: 'kept ' },
    { subject: 'the horse and the hound and the horn', action: 'belonged to ' }
  ];

  static verse(n) {
    const reversedParts = House.parts.slice(0, n).reverse();
    let lines = [`This is ${reversedParts[0].subject}`];
    for (let i = 0; i < reversedParts.length - 1; i++) {
      lines.push(`that ${reversedParts[i].action}${reversedParts[i + 1].subject}`);
    }
    return lines;
  }

  static verses(start, end) {
    let result = [];
    for (let i = start; i <= end; i++) {
      const verseLines = House.verse(i);
      result.push(...verseLines);
      if (i !== end) {
        result.push('');
      }
    }
    return result;
  }
}

File: src/2025-02-23-deepclaude-try/java/exercises/practice/house/src/main/java/House.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class House {
    private static final String[] PARTS = {
        "the house that Jack built.",
        "the malt\nthat lay in ",
        "the rat\nthat ate ",
        "the cat\nthat killed ",
        "the dog\nthat worried ",
        "the cow with the crumpled horn\nthat tossed ",
        "the maiden all forlorn\nthat milked ",
        "the man all tattered and torn\nthat kissed ",
        "the priest all shaven and shorn\nthat married ",
        "the rooster that crowed in the morn\nthat woke ",
        "the farmer sowing his corn\nthat kept ",
        "the horse and the hound and the horn\nthat belonged to "
    };

    public String verse(int verse) {
        if (verse < 1 || verse > 12) {
            throw new IllegalArgumentException("Verse must be between 1 and 12");
        }
        
        List<String> selectedParts = new ArrayList<>();
        for (int i = 0; i < verse; i++) {
            selectedParts.add(PARTS[i]);
        }
        Collections.reverse(selectedParts);
        
        StringBuilder sb = new StringBuilder("This is ");
        for (int i = 0; i < selectedParts.size(); i++) {
            if (i > 0) {
                sb.append("\n");
            }
            sb.append(selectedParts.get(i));
        }
        return sb.toString();
    }

    public String verses(int startVerse, int endVerse) {
        StringBuilder sb = new StringBuilder();
        for (int v = startVerse; v <= endVerse; v++) {
            if (v > startVerse) {
                sb.append("\n\n");
            }
            sb.append(verse(v));
        }
        return sb.toString();
    }

    public String sing() {
        return verses(1, 12);
    }
}
